File: 6/A.cpp
#include <bits/stdc++.h>

using namespace std;

int main(void)
{
    fstream fin("input.txt");

    int *fishes = new int[100000000];

    int size = 0;
    while (fin.good())
    {
        string fish;
        getline(fin, fish, ',');
        fishes[size++] = stoi(fish);
    }

    int days = 80;
    while (days--)
    {
        for (int i = 0; i < size; i++)
        {
            if (fishes[i])
                fishes[i]--;
            else
            {
                fishes[i] = 6;
                fishes[size++] = 9; // Size++, so newly added fish gets decremented
            }
        }
    }

    cout << size << endl;

    delete[] fishes;
}

File: 6/A.py
def num_fishes(fish, days):
    if days == 0:
        return 0

    if fish == 0:
        return 1 + num_fishes(6, days - 1) + num_fishes(8, days - 1)

    return num_fishes(fish - 1, days - 1)


with open("input.txt", "r") as f:
    fishes = list(map(int, f.read().split(',')))
    print(sum(map(lambda fish: num_fishes(fish, 80) + 1, fishes)))
